sql/moz-fx-data-shared-prod/telemetry_derived/clients_first_seen_v2/query.sql

```diff
@@ -0,0 +1,643 @@
+      HAVING
+        MIN submission_timestamp
+    ) AS attribution_experiment,
+    ANY_VALUE(
+      environment.settings.attribution.medium
+      HAVING
+        MIN submission_timestamp
+    ) AS attribution_medium,
+    ANY_VALUE(
+      environment.settings.attribution.source
+      HAVING
+        MIN submission_timestamp
+    ) AS attribution_source,
+    ANY_VALUE(
+      environment.settings.default_search_engine_data.load_path
+      HAVING
+        MIN submission_timestamp
+    ) AS engine_data_load_path,
+    ANY_VALUE(
+      environment.settings.default_search_engine_data.name
+      HAVING
+        MIN submission_timestamp
+    ) AS engine_data_name,
+    ANY_VALUE(
+      environment.settings.default_search_engine_data.origin
+      HAVING
+        MIN submission_timestamp
+    ) AS engine_data_origin,
+    ANY_VALUE(
+      environment.settings.default_search_engine_data.submission_url
+      HAVING
+        MIN submission_timestamp
+    ) AS engine_data_submission_url,
+    ANY_VALUE(environment.system.apple_model_id HAVING MIN submission_timestamp) AS apple_model_id,
+    ANY_VALUE(metadata.geo.city HAVING MIN submission_timestamp) AS city,
+    ANY_VALUE(metadata.geo.db_version HAVING MIN submission_timestamp) AS db_version,
+    ANY_VALUE(metadata.geo.subdivision1 HAVING MIN submission_timestamp) AS subdivision1,
+    ANY_VALUE(normalized_channel HAVING MIN submission_timestamp) AS normalized_channel,
+    ANY_VALUE(normalized_country_code HAVING MIN submission_timestamp) AS country,
+    ANY_VALUE(normalized_os HAVING MIN submission_timestamp) AS normalized_os,
+    ANY_VALUE(normalized_os_version HAVING MIN submission_timestamp) AS normalized_os_version,
+    ANY_VALUE(
+      payload.processes.parent.scalars.startup_profile_selection_reason
+      HAVING
+        MIN submission_timestamp
+    ) AS startup_profile_selection_reason,
+    ANY_VALUE(
+      environment.settings.attribution.dltoken
+      HAVING
+        MIN submission_timestamp
+    ) AS download_token,
+    ANY_VALUE(
+      environment.settings.attribution.dlsource
+      HAVING
+        MIN submission_timestamp
+    ) AS download_source
+  FROM
+    `moz-fx-data-shared-prod.telemetry.first_shutdown`
+  WHERE
+    {% if is_init() %}
+      DATE(submission_timestamp) >= '2010-01-01'
+      {% if parallel_run() %}
+        AND sample_id = {sample_id }
+      {% endif %}
+    {% else %}
+      DATE(submission_timestamp) = @submission_date
+    {% endif %}
+  GROUP BY
+    client_id,
+    sample_id
+),
+main_ping AS (
+  SELECT
+    client_id AS client_id,
+    sample_id AS sample_id,
+    IF(
+      ANY_VALUE(submission_date HAVING MIN submission_date) >= '2018-10-30',
+      MIN(submission_timestamp_min),
+      TIMESTAMP(ANY_VALUE(submission_date HAVING MIN submission_date))
+    ) AS first_seen_date,
+    TIMESTAMP(
+      ARRAY_AGG(submission_date IGNORE NULLS ORDER BY submission_date)[SAFE_OFFSET(1)]
+    ) AS second_seen_date,
```

Comment:
I'm probably missing something, but why are we using the second main ping sent day (`safe_offset(1)`) for the second seen date in this part of the query, when we're using nulls for the new profile and first shutdown ping parts? 

is this assuming that the second main ping day is always the second seen date? 

maybe I'm misunderstanding, but is second seen date supposed to be the second day we get main ping from user, by definition? Or is it supposed to be second day we get any ping from user? (definition says second day we get any which ping)

Reply:
Good question. This is the result of a discussion with Ruoxi @rhythmofrain63 where we decided to exclude new_profile and first_shutdown pings to retrieve the second seen dates because the new_profile ping is being sent more than once for around ~16% of clients (the expectation is once per client_id) and this could result in both 1st and 2nd seen dates being reported by the new_profile ping.
Additionally, we think that it doesn't make sense if the new_profile ping reports the second_seen date after the main ping already reported the first (the browser is already installed and opened), what do you think?

Reply:
My 2c, I think it makes sense to exclude the new profile ping from contributing the second_seen_date. Assuming the intent here is to capture their second day of use, receiving the NPP after their first main ping would not be a fair reflection of their second day (bcs its likely we SHOULD have actually received the NPP first, and we are just getting it second for uninteresting reasons). 

Technically, since the first_shutdown_ping might often be sent on the next UTC day, we might consider that their "second seen date", similarly to how we would expect a second main ping from them on their next UTC day, if they leave the browser open. So that would be an argument for allowing the FSP to define the second_seen_date. That being said, I'm not sure if that _really_ buys us all that much more in exchange for the added complexity.

On balance, I think that we should probably stick with the idea that your second_seen date is the second UTC day we receive a main ping from each client, if such a second day occurs.